src/NetAnalyzers/UnitTests/Microsoft.CodeQuality.Analyzers/QualityGuidelines/UseLiteralsWhereAppropriateTests.cs

```diff
@@ -169,6 +169,26 @@ End Class
             await vbTest.RunAsync();
         }
 
+        [Fact]
+        public async Task CA1802_CSharp_IntPtr_UIntPtr_NoDiagnostic()
+        {
+            await new VerifyCS.Test
+            {
+                LanguageVersion = CodeAnalysis.CSharp.LanguageVersion.CSharp9,
+                ReferenceAssemblies = ReferenceAssemblies.Net.Net50,
+                TestCode = @"
+using System;
+
+public class Class1
+{
+	internal static readonly IntPtr field1 = (nint)0;
+	internal static readonly UIntPtr field2 = (nuint)0;
+    internal static readonly nint field3 = (nint)0;
+	internal static readonly nuint field4 = (nuint)0;
```

Comment:
❕ These two lines _should_ have a diagnostic

Reply:
➡️ This discussion covered by https://github.com/dotnet/roslyn-analyzers/pull/4550#discussion_r540266278